specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/ContainerAppsPatches.json

```diff
@@ -0,0 +1,563 @@
+      },
+      "readOnly": true
+    },
+    "PatchApplyStatus": {
+      "type": "string",
+      "description": "The status of the patch once it has been provisioned",
+      "enum": [
+        "NotStarted",
+        "RebaseInProgress",
+        "CreatingRevision",
+        "Succeeded",
+        "Canceled",
+        "RebaseFailed",
+        "RevisionCreationFailed",
+        "ImagePushPullFailed",
+        "ManuallySkipped"
+      ],
+      "x-ms-enum": {
+        "name": "PatchApplyStatus",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "DetectionStatus": {
+      "type": "string",
+      "description": "The status of the patch detection.",
+      "enum": [
+        "Succeeded",
+        "RegistryLoginFailed",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "name": "DetectionStatus",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "PatchDetails": {
+      "description": "The detailed info of patch operation performing when applying a patch.",
+      "type": "object",
+      "required": [
+        "targetContainerName",
+        "targetImage",
+        "lastDetectionTime",
+        "detectionStatus"
+      ],
+      "properties": {
+        "targetContainerName": {
+          "type": "string",
+          "description": "The name of the target container for the patch.",
+          "readOnly": true
+        },
+        "targetImage": {
+          "type": "string",
+          "description": "The name of the target image for the patch.",
+          "readOnly": true
+        },
+        "lastDetectionTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The UTC timestamp that describes the latest detection was done.",
+          "readOnly": true
+        },
+        "detectionStatus": {
+          "$ref": "#/definitions/DetectionStatus",
+          "description": "The status of the patch detection.",
+          "readOnly": true
+        },
+        "newImageName": {
+          "type": "string",
+          "description": "The name of the new image created by the patch.",
+          "readOnly": true
+        },
+        "newLayer": {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string",
+              "description": "The details of the new layer for the target image."
+            },
+            "frameworkAndVersion": {
+              "type": "string",
+              "description": "The framework and its version in the new run image for the target image."
+            },
+            "osAndVersion": {
+              "type": "string",
+              "description": "The OS name and its version in the new run image for the target image."
+            }
+          },
+          "description": "New layer update details in the target image.",
+          "readOnly": true
+        },
+        "oldLayer": {
+          "type": "object",
+          "properties": {
```

Comment:
Consider using `x-ms-client-flatten` for nested properties: https://github.com/Azure/azure-openapi-validator/blob/main/docs/avoid-nested-properties.md

Reply:
I don't think you want to use flatten here - your newLayer and oldLayer property names will collide if you do.